src/main/scala/nl.knaw.dans.easy.multideposit/actions/AddDatasetMetadataToDeposit.scala

```diff
@@ -23,11 +23,35 @@ import org.joda.time.format.ISODateTimeFormat
 
 import scala.language.postfixOps
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Try }
+import scala.util.{ Failure, Success, Try }
 import scala.xml.{ Elem, Null, PrefixedAttribute }
 
 case class AddDatasetMetadataToDeposit(deposit: Deposit)(implicit settings: Settings) extends UnitAction[Unit] {
 
+  /**
+   * Verifies whether all preconditions are met for this specific action.
+   *
+   * @return `Success` when all preconditions are met, `Failure` otherwise
+   */
+  override def checkPreconditions: Try[Unit] = {
+    def checkSpringFieldDepositHasAVformat(): Try[Unit] = {
+
+      (deposit.audioVideo.springfield.isDefined, deposit.metadata.formats.filter(s => s.startsWith("audio/")||s.startsWith("video/")).isEmpty) match {
+        case(false, _) => Success(())
+        case(true, false) => Success(())
+        case(true, true) =>
+          Failure(ActionException(deposit.row,
+            "No audio/video Format found for this column: [DC_FORMAT]\n" +
+              "cause: this column should contain at least one " +
+              "audio/ or video/ value because SF columns are present"))
+      }
+    }
+
+    for {
```

Comment:
no need for a for-comprehension here. If you only want a `()` at the end, just use `map`. Then this call can be a one-liner.